pkg/js/transformer/modules/modules-to-model.ts

```diff
@@ -0,0 +1,191 @@
+  // Stitch them into the main model
+  const model: Omit<AuthorizationModel, "id"> = {
+    schema_version: "1.2",
+    type_definitions: [],
+    conditions: {}
+  };
+
+  const typeDefs: TypeDefinition[] = [];
+  const types = new Set<string>();
+  const extendedTypeDefs: Record<string, TypeDefinition[]> = {};
+  const conditions = new Map<string, Condition>();
+  const errors: BaseError[] = [];
+  const moduleFiles = new Map(files.map(file => [file.name, file.contents]));
+
+  for (const { name, contents } of files) {
+    try {
+      const lines = contents.split("\n");
+      const { authorizationModel, typeDefExtensions } = transformModularDSLToJSONObject(contents);
+
+      for (const typeDef of authorizationModel.type_definitions) {
+        // Check if we've already seen this type and it's not marked as an extension
+        if (types.has(typeDef.type) && !typeDefExtensions.has(typeDef.type)) {
+          const lineIndex = getTypeLineNumber(typeDef.type, lines);
+          errors.push(constructTransformationError({
+            message: `duplicate type definition ${typeDef.type}`,
+            lines, lineIndex,
+            metadata: {
+              symbol: typeDef.type,
+              file: name,
+            }
+          }));
+          continue;
+        }
+
+        // If this is an extension mark it to be merged later
+        if (typeDefExtensions.has(typeDef.type)) {
+          if (!extendedTypeDefs[name]) {
+            extendedTypeDefs[name] = [];
+          }
+          extendedTypeDefs[name].push(typeDef);
+          continue;
+        }
+
+        typeDef.metadata!.file = name;
+        types.add(typeDef.type);
+        typeDefs.push(typeDef);
+      }
+
+      if (authorizationModel.conditions) {
+        for (const [conditionName, condition] of Object.entries(authorizationModel.conditions)) {
+          // If we have already seen a condition with this name mark it as duplicate
+          if (conditions.has(conditionName)) {
+            const lineIndex = getConditionLineNumber(conditionName, lines);
+            errors.push(constructTransformationError({
+              message: `duplicate condition ${conditionName}`,
+              lines, lineIndex,
+              metadata: {
+                symbol: conditionName,
+                file: name,
+              }
+            }));
+            continue;
+          }
+          condition.metadata!.file = name;
+          conditions.set(conditionName, condition);
+        }
+      }
+    } catch (error) {
+      if (error instanceof DSLSyntaxError) {
+        errors.push(...error.errors);
+      } else if (error instanceof Error) {
+        errors.push(error as BaseError);
+      }
+    }
+  }
+
+  for (const [filename, extended] of Object.entries(extendedTypeDefs)) {
+    const lines = moduleFiles.get(filename)?.split("\n");
+    for (const typeDef of extended) {
+
+      if (!typeDef.relations) {
+        // TODO: Maybe should be an error case or at least a warning?
```

Comment:
That's a good question - I think in the future we may want to allow it, but for not, it makes sense not to allow type extensions that do nothing